American Option Dynamic Prog.cpp

```diff
@@ -0,0 +1,148 @@
+// Calculating the price of an American Option using Dynamic Programming
+//where the Probability Matrix is derived from the Trinomial Lattice.
+
+#include <iostream>
+#include <iomanip>
+#include <cmath>
+#include <fstream>
+#include <cstdlib>
+#include <algorithm>
+#include ".../newmat10/newmat.h"
+using namespace std;
+
+float up_factor, uptick_prob, downtick_prob, notick_prob, risk_free_rate, strike_price;
+float initial_stock_price, expiration_time, volatility, R;
+int no_of_divisions;
+
+float max(float a, float b) {
+	return (b < a )? a:b;
+}
+
+float american_call_option_dyn_prog()
+{
+
+	// create the probability matrix
+	Matrix transition_probability(2*no_of_divisions+1, 2*no_of_divisions+1);
```

Comment:
Modify the constructor of Matrix, let the default constructor create an empty matrix